hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -453,7 +453,12 @@ abstract class RegistryFunctions {
 
       override val returnType: Type = rType
 
-      override def returnPType(argTypes: Seq[PType], returnType: Type): PType = if (pt == null) PType.canonical(returnType) else pt(argTypes)
+      override def returnPType(argPTypes: Seq[PType], returnType: Type): PType = {
```

Comment:
interesting, this change breaks test/hail/methods/test_statgen.py::Tests::test_ld_prune_no_prune - hail.utils.java.FatalError: HailException: block matrix must have at least one row

Reply:
that's not good.

Reply:
tracking it, here are the mnames and returnTypes leading up to the error:

```
returnType is int32 and mName is rand_cat
returnType is int32 and mName is rand_cat
returnType is float64 and mName is rand_unif
returnType is float64 and mName is rand_beta
returnType is int32 and mName is rand_cat
returnType is float64 and mName is rand_unif
returnType is float64 and mName is rand_beta
returnType is int32 and mName is rand_cat
returnType is int32 and mName is rand_cat
returnType is int32 and mName is rand_cat
returnType is float64 and mName is rand_unif
returnType is float64 and mName is rand_beta
returnType is int32 and mName is rand_cat
returnType is float64 and mName is rand_unif
returnType is float64 and mName is rand_beta
returnType is int32 and mName is rand_cat
```

will resume after 2:30pm

Reply:
can be replicated by setting rand_cat's requirement based on the requiredness of the input PCArray, which is +PCArray[PFloat64]